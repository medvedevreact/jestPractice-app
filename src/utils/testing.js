import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

export const renderWithRouter = (component, options = {}) => {
  const Wrapper = ({ children }) => (
    <MemoryRouter {...options}>{children}</MemoryRouter>
  );
  return render(component, { wrapper: Wrapper });
};
