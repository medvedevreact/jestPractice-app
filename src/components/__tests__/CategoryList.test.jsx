import { screen } from '@testing-library/react';
import { CategoryList } from '../CategoryList';
import { renderWithRouter } from '../../utils/testing';

describe('CategoryList component', () => {
  const categoriesMock = [
    {
      idCategory: '1',
      strCategory: 'Dessert',
      strCategoryThumb: '/dessert.png',
      strCategoryDescription: 'A variety of delicious desserts.',
    },
    {
      idCategory: '2',
      strCategory: 'Main Course',
      strCategoryThumb: '/main-course.png',
      strCategoryDescription: 'Delicious main course meals.',
    },
  ];

  it('renders the list of categories', () => {
    renderWithRouter(<CategoryList catalog={categoriesMock} />);

    const list = screen.getByRole('list');
    expect(list).toBeInTheDocument();

    const items = screen.getAllByRole('article');
    expect(items).toHaveLength(categoriesMock.length);
  });

  it('renders nothing when catalog is empty', () => {
    renderWithRouter(<CategoryList />);

    const list = screen.queryByRole('list');
    expect(list).toMatchSnapshot();
  });
});
