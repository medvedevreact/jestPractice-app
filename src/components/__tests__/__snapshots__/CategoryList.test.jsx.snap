// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryList component renders nothing when catalog is empty 1`] = `
<div
  class="list"
  role="list"
/>
`;
